src/vs/workbench/parts/debug/electron-browser/debugService.ts

```diff
@@ -1042,9 +1042,11 @@ export class DebugService implements debug.IDebugService {
 		}
 		// run a task before starting a debug session
 		return this.taskService.getTask(root, taskId).then(task => {
-			const taskDisplayName = typeof taskId === 'string' ? `'${taskId}'` : nls.localize('specified', "specified");
 			if (!task) {
-				return TPromise.wrapError(errors.create(nls.localize('DebugTaskNotFound', "Could not find the task {0}.", taskDisplayName)));
+				if (typeof taskId === 'string')
```

Comment:
Use brackets {} after an if
Also based on the state construct the message and then in bot hcases just wrap error. Same as you do below